deploy/openshift/tools/contrail-sc-open/main.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"flag"
+	"log"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+type port struct {
+	From, To int64
+	Protocol string
+}
+
+const (
+	protocolTCP = "tcp"
+	protocolUDP = "udp"
+)
+
+// ports is an array of all port ranges that should be open in order to make contrail work properly
+var ports = [...]port{
+	{5995, 5995, protocolTCP},
+	{6379, 6379, protocolTCP},
+	{5920, 5920, protocolTCP},
+	{5921, 5921, protocolTCP},
+	{4369, 4369, protocolTCP},
+	{5673, 5673, protocolTCP},
+	{25672, 25672, protocolTCP},
+	{514, 514, protocolTCP},
+	{6343, 6343, protocolTCP},
+	{4739, 4739, protocolTCP},
+	{5269, 5269, protocolTCP},
+	{53, 53, protocolTCP},
+	{179, 179, protocolTCP},
+	{5672, 5672, protocolTCP},
+	{10250, 10250, protocolTCP},
+	{10256, 10256, protocolTCP},
+	{80, 80, protocolTCP},
+	{443, 443, protocolTCP},
+	{1936, 1936, protocolTCP},
+	{7000, 10000, protocolTCP},
+	{2000, 3888, protocolTCP},
+	{8053, 8053, protocolUDP},
+	{4789, 4789, protocolUDP},
+}
+
+func setRules(svc *ec2.EC2, group *string) error {
+	ruleCIDR := "0.0.0.0/0"
+	for portIndex := range ports {
+		_, err := svc.AuthorizeSecurityGroupIngress(&ec2.AuthorizeSecurityGroupIngressInput{
+			GroupId: group,
+			IpPermissions: []*ec2.IpPermission{
+				&ec2.IpPermission{
+					IpProtocol: &ports[portIndex].Protocol,
+					FromPort:   &ports[portIndex].From,
+					ToPort:     &ports[portIndex].To,
+					IpRanges: []*ec2.IpRange{
+						&ec2.IpRange{
+							CidrIp: &ruleCIDR,
+						},
+					},
+				},
+			},
+		})
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
```

Comment:
Duplicate is the only situation where you want to throw an error?
If there is a risk that some other errors also should be reported then instead of throwing error if it matches certain code DON'T throw an error if it matches certain code.

Reply:
Duplicate is the only situation where i don't want to throw an error ;)

Reply:
Duplicate is the only situation where i don't want to throw an error ;)

Reply:
Done